lib/instrumentation/http-shared.js

```diff
@@ -189,7 +190,16 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
 
         function onEnd () {
           agent.logger.debug('intercepted http.IncomingMessage end event %o', { id: id })
-          span.end()
+
+          httpRequestToUrl(req).then(url => {
```

Comment:
Isn't there a risk that this promise will never resolve in case `req.socket` is never populated for some reason?

Reply:
Yep, there was also some test issues due to the timing, so I changed it. The url is fetched out-of-band now, so it'd just be `undefined` if the socket is never populated.